dialog/file.go

```diff
@@ -245,27 +319,122 @@ func showFileDialog(save bool, callback interface{}, parent fyne.Window) {
 	size := ui.MinSize().Add(fyne.NewSize(fileIconCellWidth*2+theme.Padding()*4,
 		(fileIconSize+fileTextSize)+theme.Padding()*4))
 
-	d.win = widget.NewModalPopUp(ui, parent.Canvas())
+	d.win = widget.NewModalPopUp(ui, file.parent.Canvas())
 	d.win.Resize(size)
 
 	d.win.Show()
+	return d
+}
+
+// Show shows the file dialog.
+func (f *FileDialog) Show() {
+	if f.save {
+		if fileSaveOSOverride(f.callback.(func(fyne.FileWriteCloser, error)), f.parent) {
+			f.nativePicker = true
+			return
+		}
+	} else {
+		if fileOpenOSOverride(f.callback.(func(fyne.FileReadCloser, error)), f.parent) {
+			f.nativePicker = true
+			return
+		}
+	}
+	if f.dialog != nil {
+		f.dialog.win.Show()
+		return
+	}
+	f.dialog = showFile(f)
+}
+
+// Hide hides the file dialog.
+func (f *FileDialog) Hide() {
+	if f.nativePicker {
+		return
+	}
+	f.dialog.win.Hide()
+	if f.onClosedCallback != nil {
+		f.onClosedCallback(false)
+	}
+}
+
+// SetDismissText allows custom text to be set in the confirmation button
+func (f *FileDialog) SetDismissText(label string) {
+	if f.nativePicker {
+		return
+	}
+	f.dialog.dismiss.SetText(label)
+	widget.Refresh(f.dialog.win)
+}
+
+// SetOnClosed sets a callback function that is called when
+// the dialog is closed.
+func (f *FileDialog) SetOnClosed(closed func()) {
+	// If there is already a callback set, remember it and call both.
```

Comment:
I think you added code in the wrong place so this document no longer applies to the code it's next to